docs/CODE_GUIDELINES.md

```diff
@@ -280,82 +415,200 @@ enum Dummy
 }
 ```
 
-### 11.5. Classes
-Use CamelCase for class names and they have to be prefixed with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
+### 8.5. Classes
+Use PascalCase and prefix with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
 ```cpp
 class CLogger : public ILogger
 {
-  void Log(...)
+public:
+  virtual void Log(...) override;
 }
 ```
 
-### 11.6. Methods
-Use CamelCase for method names and first letter has to be uppercase, even if the methods are private or protected.
+### 8.6. Methods
+Use PascalCase always, uppercasing the first letter even if the methods are private or protected.
 ```cpp
 void MyDummyClass::DoSomething();
 ```
 
-### 11.7. Variables
-Use CamelCase for variables. Type prefixing is discouraged.
+### 8.7. Variables
+Use CamelCase. Type prefixing (Systems Hungarian notation) is discouraged.
+
+#### Member variables
+Prefix nonstatic member variables with `m_`. Prefix static member variables with `ms_`.
+```cpp
+int m_variableA;
+static int ms_variableB;
+```
 
-#### Global Variables
-Prefix global variables with *g_*
+#### Global variables
+Prefix global variables with `g_`
 ```cpp
 int g_globalVariableA;
 ```
-**WARNING:** Avoid globals use. It increases the chances of submitted code to be rejected.
+**WARNING:** Avoid use of globals as far as reasonably possible. We generally do not want to introduce new global variables.
+
+**[back to top](#table-of-contents)**
 
-#### Member Variables
-Prefix member variables with *m_*
+## 9. Comments
+
+### 9.1. General
+
+Use `// ` for inline single-line and multi-line comments. Use `/* */` for the copyright comment at the beginning of the file. SPDX license headers are required for all code files (see example below).
+
+✅ Good:
 ```cpp
-int m_variableA;
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+// Nice comment
+
+// This can also continue for multiple lines:
+// I am the second line.
+```
+
+❌ Bad:
+```cpp
+/* A comment */
+//another comment
+```
+
+### 9.2. Doxygen
+
+New classes and functions are expected to have Doxygen comments describing their purpose, parameters, and behavior in the header file. However, do not describe trivialities - it only adds visual noise. Use the Qt style with exclamation mark (`/*! */`) and backslash for doxygen commands (e.g. `\brief`). 
+
+✅ Good:
+```cpp
+/*!
+ * \brief Set decoration state and size by providing full surface size
+ *        including decorations
+ *
+ * Calculates size of the main surface from size of all surfaces combined (including
+ * window decorations) by subtracting the decoration size
+ *
+ * Decorations will be disabled if state includes STATE_FULLSCREEN
+ *
+ * Call only from main thread
+ */
+void SetState(CSizeInt size, int scale, IShellSurface::StateBitset state);
```

Comment:
Maybe include an example where the parameters and return type are doxy'd?

Reply:
> Maybe include an example where the parameters and return type are doxy'd?

Good idea. Do you have one in mind?

> I saw below, I think that parameters should be explicitly documented, even if it's trivial (though trivial parameter descriptions are discouraged, of course)

Not completely sure, but I tend to disagree. Comments should only be there if they add value. What value do you think is added by having a doxy comment like in the (intentionally bad) example?

Reply:
Messaging/helpers should have parameters and return types documented.
Appmessenger and surrounding code is well documented (quality is debatable 😀)

Reply:
In this case, my argument isn't for the result but the habit - developers should practice documenting everything. Oftentimes, I'll document three parameters, the first two will be trivial but the third I expand upon. The "noise" from the first two is worth the reminder to myself to expand upon the more opaque parameter.

It's like including the many headers in our PR templates - the dev is free to ignore and delete any headers, but I want them to be conscious that they're doing so. It should take extra effort to document less, not the other way around.

Reply:
> Messaging/helpers should have parameters and return types documented.

Tried a utils one.

Related question: Do we want to have the `*` on the left side of the doxy comments? I kind of like them because it makes it more visually distinct in the editor, but I'm open for opinions.

> In this case, my argument isn't for the result but the habit - developers should practice documenting everything.

I'm seeing the opposite problem - if you get into the habit of "documenting" everything by adding boilerplate text that does not add information, you not only clutter the code, but may also grow into an automatism that prevents you from writing actually meaningful comments.